Blackjack/Model/GamerResult.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  GamerResult.swift
+//  Blackjack
+//
+//  Created by 강수진 on 2022/05/25.
+//
+
+import Foundation
+
+struct GamerResult {
+    
+    enum Outcome: UserInformable {
+        case win
+        case draw
+        case lose
+        
+        var guideDescription: String {
+            switch self {
+            case .win:
+                return "승"
+            case .draw:
+                return "무"
+            case .lose:
+                return "패"
+            }
+        }
+    }
+    
+    private let dealer: Dealer
+    private let gamers: [Gamer]
+    private var winningPoint: Int {
+        let allPlayers: [Gamer] = gamers + [dealer]
+        let winningPoint: Int = allPlayers
+            .filter { gamer in
+                ThresholdChecker().isTotalPointUnderThreshold(of: gamer.cards)
+            }
+            .map { $0.totalPoint }.max() ?? 0
+        return winningPoint
+    }
+    
+    init(dealer: Dealer, gamers: [Gamer]) {
+        self.dealer = dealer
+        self.gamers = gamers
+    }
+    
+    func outcome(of gamer: Gamer) -> Outcome {
+        guard !dealer.isBurst else {
+            return .win
+        }
+        
+        guard !gamer.isBurst else {
+            return .lose
+        }
+        
+        let isGamerPointOverWinningPoint: Bool = gamer.totalPoint >= winningPoint
+        if isGamerPointOverWinningPoint {
+            let isGamerPointSameWithDealer: Bool = dealer.totalPoint == gamer.totalPoint
+            if isGamerPointSameWithDealer {
+                return .draw
+            } else {
+                return .win
+            }
+        } else {
+            return .lose
+        }
```

Comment:
if 안에 또 if 가 있어서 썩 맘에 들진 않는데..! 어떻게 고쳐야할지 떠오르는게 없네여 🤔🤔🤔🤔🤔
혹시 제안 주실 내용 있으실까요?? 🫠

Reply:
```suggestion
    let isGamerPointSameWithDealer: Bool = dealer.totalPoint == gamer.totalPoint
    if !isGamerPointOverWinningPoint {
      return .lose
    }
    if isGamerPointSameWithDealer {
      return .draw
    } else {
      return .win
    }
```
if문 중첩을 한번 없애보았는데, 원하는 방향이 맞으실지 모르겠네요 🤔

Reply:
depth 가 줄어서 좀 더 나은것 같네요 👍🏻
말씀주신 방향 참고해서 로직 수정했습니다
https://github.com/next-step/swift-blackjack/pull/10/commits/b6275ce365a1c73cd3507419f88a9f67ac3d37ef